### PR TITLE
feat: create sm datasource context

### PR DESCRIPTION
**Pull request by ckbedwell**
## Problem

The Synthetic Monitoring datasource is key in being accessible to be able to use the Synthetic Monitoring app plugin. Currently access to it lives in the instances context alongside the metrics and logs datasources, however we don't know what metrics and logs datasources to use until we have access to the Synthetic Monitoring datasource.

## Solution

Move the retrieval of the Synthetic Monitoring Datasource to its own context that lives as a parent to the Logs/Metrics datasources. This is the second item in the RBAC task list: https://github.com/grafana/synthetic-monitoring-app/issues/877

### CODE REVIEW

File: src/components/Routing.tsx [hunk 1]
```diff
@@ -108,7 +109,7 @@ export const UninitialisedRouter = () => {
         <AlertingWelcomePage />
       </Route>
       <Route path={getRoute(ROUTES.Config)}>
-        <ConfigPage />
+        {hasRole(OrgRole.Editor) ? <ConfigPage /> : <Redirect to={getRoute(ROUTES.Home)} />}
```
Comment (ckbedwell) [inline src/components/Routing.tsx:112]: ❤️ 💌  @VikaCep for tests otherwise I would have missed adding these back in 💌❤️

**Comment by VikaCep** [comment]
Not sure if there is something missing in my env but this is how I see the config section:

image

File: src/contexts/SMDatasourceContext.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+type SMDatasourceContextValue = {
+  smDS: SMDataSource;
+} | null;
+
+export const SMDatasourceContext = createContext<SMDatasourceContextValue>(null);
+
+export const SMDatasourceProvider = ({ children }: PropsWithChildren) => {
+  const { data, isLoading } = useGetSMDatasource();
+
+  if (isLoading && !data) {
```
[the remark this answers is not in the document]
Comment (ckbedwell) [reply src/contexts/SMDatasourceContext.tsx]: Not in this case. I've removed it, good spot 😄

**Comment by ckbedwell** [comment]
> Not sure if there is something missing in my env but this is how I see the config section:
> 
> image

Good catch! I've added tests now so this will get caught in the future, too 😄

**Review by VikaCep — APPROVED** [review]
💯💯💯

**Pull request merged**